static/js/components/AssignmentForm.jsx

```diff
@@ -0,0 +1,158 @@
+import { makeStyles } from "@material-ui/core/styles";
+import * as Actions from '../ducks/source';
+
+import 'react-datepicker/dist/react-datepicker-cssmodules.css';
+
+export function observingRunTitle(observingRun, instrumentList, telescopeList, groups) {
+  const { instrument_id } = observingRun;
+  const instrument = instrumentList?.filter((i) => i.id === instrument_id)[0];
+
+  const telescope_id = instrument?.telescope_id;
+  const telescope = telescopeList?.filter((t) => t.id === telescope_id)[0];
+
+  const group = groups?.filter((g) => g.id === observingRun.group_id)[0];
+
+  return `${observingRun?.calendar_date} ${instrument?.name}/${telescope?.nickname} (PI: ${observingRun?.pi} ${group?.name})`;
+}
+
+function makeMenuItem(observingRun, instrumentList, telescopeList, groups) {
+  const render_string = observingRunTitle(observingRun, instrumentList, telescopeList, groups);
+  return (
+    <MenuItem value={observingRun.id} key={observingRun.id.toString()}>
+      {render_string}
+    </MenuItem>
+  );
+}
+
+const AssignmentForm = ({ obj_id, observingRunList }) => {
+  const dispatch = useDispatch();
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const groups = useSelector((state) => state.groups.all);
+
+  const upcomingRuns = observingRunList.filter((observingrun) => (
+    Date.parse(observingrun.ephemeris.sunrise_utc) >= Date.now()
+  ));
+
+  const { handleSubmit, getValues, reset, register, control } = useForm();
+
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120
+    }
+  }));
+  const classes = useStyles();
+
+  if (upcomingRuns.length === 0) {
+    return (
+      <b>
+        No upcoming observing runs to assign target to...
+      </b>
+    );
+  }
+
+  const initialFormState = {
+    comment: "",
+    run_id: upcomingRuns.length > 0 ? upcomingRuns[0].id : null,
+    priority: "1",
+    obj_id
+  };
+
+  const onSubmit = () => {
+    const formData = {
+      // Need to add obj_id, etc to form data for request
+      ...initialFormState,
+      ...getValues()
+    };
+    dispatch(Actions.submitAssignment(formData));
+    reset(initialFormState);
+  };
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <h3>
+          Assign Target to Observing Run
+        </h3>
+        <div>
+          <FormControl className={classes.formControl}>
+            <InputLabel id="assignmentSelectLabel">
+              Choose Run
+            </InputLabel>
+            <Controller
+              as={Select}
+              labelId="assignmentSelectLabel"
+              name="run_id"
+              control={control}
+              rules={{ required: true }}
+              defaultValue={upcomingRuns.length > 0 ? upcomingRuns[0].id : null}
+            >
+              {upcomingRuns.map((observingRun) => makeMenuItem(
```

Comment:
Please remove `makeMenuItem` above and just move those few lines of JSX to here (and just call observingRunTitle in between the opening and closing MenuItem tags, no need to create that intermediate as in that function def above)

Reply:
Done